src/css.js

```diff
@@ -58,13 +58,31 @@ function vendorPropName( name ) {
 	}
 }
 
+// Use font-size to convert to px
+function convertToPx( elem, value ) {
+	var fontSize = jQuery.css( elem, "font-size" );
+	elem.style.fontSize = value;
+	value = parseFloat( jQuery.css( elem, "font-size" ) );
+	elem.style.fontSize = fontSize;
+	return value;
+}
+
 function setPositiveNumber( elem, value, subtract ) {
-	var matches = rnumsplit.exec( value );
-	return matches ?
+	var unit,
+		matches = rnumsplit.exec( value );
+	if ( matches ) {
+		unit = matches[ 2 ] || "px";
+
+		// Only need to convert if we're manipulating the value
+		if ( unit !== "px" && subtract ) {
+			matches[ 1 ] = convertToPx( elem, value );
```

Comment:
I _love_ this force-to-px strategy, but get the feeling that `setPositiveNumber` is the wrong place for it. What if we moved everything out into the height/width cssHooks (the only place providing `subtract` anyway), where we know which property is being set?
```js
set: function( elem, value, extra ) {
	var valueParts,
		styles = extra && getStyles( elem ),
		subtract = extra && augmentWidthOrHeight(
			elem,
			name,
			extra,
			jQuery.css( elem, "boxSizing", false, styles ) === "border-box",
			styles
		);

	// Match units if we have margin/border/padding pixels to subtract
	if ( subtract && (valueParts = rcssNum.exec( value )) && (valueParts[ 3 ] || "px") !== "px" ) {
		elem.style[ name ] = value;
		value = jQuery.css( elem, name );
	}

	return setPositiveNumber( elem, value, subtract );
}
```